common/lib/capa/capa/safe_exec/safe_exec.py

```diff
@@ -141,28 +143,48 @@ def safe_exec(
     # Create the complete code we'll run.
     code_prolog = CODE_PROLOG % random_seed
 
-    # Decide which code executor to use.
-    if unsafely:
-        exec_fn = codejail_not_safe_exec
+    if is_codejail_rest_service_enabled():
+        data = {
+            "code": code_prolog + LAZY_IMPORTS + code,
+            "globals_dict": globals_dict,
+            "python_path": python_path,
+            "limit_overrides_context": limit_overrides_context,
+            "slug": slug,
+            "unsafely": unsafely
+        }
+        response_globals_dict, emsg = send_safe_exec_request(data, extra_files)
+
+        if emsg:
```

Comment:
I think this logic of creating the exception and updating the globals should be performed by `send_safe_exec_request` method